src/Classes/Utils/ColorUtils.ts

```diff
@@ -0,0 +1,199 @@
+        const maxColor = minColor ^ 0xFFFFFF;
+        const randomColor = (Math.random() * maxColor + minColor).toString(16);
+
+        return this.stringToRgb(`#${randomColor}`) ?? {
+            b: 0,
+            g: 0,
+            r: 0,
+        };
+    }
+
+    /**
+     * Prepares a rgba() css function from a [[IRgb]] object
+     * @param color the [[IRgb]] color to convert
+     */
+    public static getStyleFromColor(color: IRgb, opacity?: number): string {
+        const opacityValue = opacity ?? 0.4;
+
+        return `rgba(${Math.round(color.r)}, ${Math.round(color.g)}, ${Math.round(color.b)}, ${opacityValue})`;
+    }
+
+    /**
+     *
+     * @param p
+     * @param q
+     * @param t
+     */
+    private static hue2rgb(p: number, q: number, t: number): number {
+        let tCalc = t;
+
+        if (tCalc < 0) {
+            tCalc += 1;
+        }
+
+        if (tCalc > 1) {
+            tCalc -= 1;
+        }
+
+        if (tCalc < 1 / 6) {
+            return p + (q - p) * 6 * tCalc;
+        }
+
+        if (tCalc < 1 / 2) {
+            return q;
+        }
+
+        if (tCalc < 2 / 3) {
+            return p + (q - p) * (2 / 3 - tCalc) * 6;
+        }
+
+        return p;
+    }
+
+    private static stringToRgba(input: string): IRgba | undefined {
+        if (input.startsWith('rgb')) {
+            const regex = /rgba?\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*(,\s*([\d\.]+)\s*)?\)/i;
+            const result = regex.exec(input);
+
+            return result ? {
+                a: parseInt(result[4]),
+                b: parseInt(result[3]),
+                g: parseInt(result[2]),
+                r: parseInt(result[1]),
+            } : undefined;
+        } else if (input.startsWith('hsl')) {
+            const regex = /hsla?\(\s*(\d+)\s*,\s*(\d+)\%\s*,\s*(\d+)\%\s*(,\s*([\d\.]+)\s*)?\)/i;
+            const result = regex.exec(input);
+
+            return result ? ColorUtils.hslaToRgba({
+                a: parseInt(result[4]),
+                h: parseInt(result[1]),
+                l: parseInt(result[3]),
+                s: parseInt(result[2]),
+            }) : undefined;
+        }
+        else {
+            // By Tim Down - http://stackoverflow.com/a/5624139/3493650
+            // Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
+            const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])([a-f\d])?$/i;
+            const hexFixed = input.replace(shorthandRegex, (_m, r, g, b, a) => {
+                return r + r + g + g + b + b + (a ? a + a : "");
+            });
+            const regex = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})?$/i;
+            const result = regex.exec(hexFixed);
+
+            return result ? {
+                a: parseInt(result[4], 16) / 0xFF,
+                b: parseInt(result[3], 16),
+                g: parseInt(result[2], 16),
+                r: parseInt(result[1], 16),
+            } : undefined;
+        }
+    }
+}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [file should end with a newline](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5216054)